src/Fields/indices.jl

```diff
@@ -0,0 +1,75 @@
+"""
+    ColumnIndex(ij,h)
+
+An index into a column of a field. This can be used as an argument to `getindex`
+of a `Field`, to return a field on that column.
+
+# Example
+```julia
+colidx = ColumnIndex((1,1),1)
+field[colidx]
+```
+"""
+struct ColumnIndex{N}
+    ij::NTuple{N, Int}
+    h::Int
+end
+
+Base.getindex(field::Field, colidx::ColumnIndex) = column(field, colidx)
+
+function column(field::SpectralElementField1D, colidx::ColumnIndex{1})
+    column(field, colidx.ij[1], colidx.h)
+end
+function column(field::ExtrudedFiniteDifferenceField, colidx::ColumnIndex{1})
+    column(field, colidx.ij[1], colidx.h)
+end
+function column(field::SpectralElementField2D, colidx::ColumnIndex{2})
+    column(field, colidx.ij[1], colidx.ij[2], colidx.h)
+end
+function column(field::ExtrudedFiniteDifferenceField, colidx::ColumnIndex{2})
+    column(field, colidx.ij[1], colidx.ij[2], colidx.h)
+end
+
+"""
+    Fields.bycolumn(fn, space)
+
+Call `fn(colidx)` to every [`ColumnIndex`](@ref) `colidx` of `space`. This can be used to apply
+multiple column-wise operations in a single pass, making use of multiple threads.
+
+# Example
+
+```julia
+∇ = GradientF2C()
+div = DivergenceC2F()
+
+bycolumn(axes(f)) do colidx
+    @. ∇f[colidx] = ∇(f[colidx])
+    @. df[colidx] = div(∇f[colidx])
+end
+```
+"""
+function bycolumn(fn, space::Spaces.SpectralElementSpace1D)
+    Nh = Topologies.nlocalelems(space)
+    Nq = Spaces.Quadratures.degrees_of_freedom(Spaces.quadrature_style(space))
+    Threads.@threads for h in 1:Nh
```

Comment:
Do we need `@inbounds`? And the `@threads` is ignored when no threading is used, correct?